### DISCUSSION TITLE
Implement methods for getting different parts of BIP32 path in address path package

### DISCUSSION THREAD

**Discussion by amirvalhalla** [pactus-project/pactus#824] [Ideas]
@b00f 
@themantre 

I think it would be a good idea to define different methods in `addresspath` package to get different parts of BIP32 path.
e.g. we defined an `AddressType` method which returns the address type from BIP32 path. [(here)](https://github.com/pactus-project/pactus/blob/main/wallet/addresspath/path.go)

we could use same approach for other parts of BIP32 path.

**Comment by b00f** [comment]
Yes, This is a good idea.
We can export function to return purpose, coin type and etc.

**Reply by amirvalhalla** [reply]
@b00f 
I think we have reached a conclusion for this discussion. can I close this discussion? also should we write a PIP for this or just implement ?

**Reply by b00f** [reply]
I don't think we need the PIP for this. Feel free to close it.
